hcloud/cloud.go

```diff
@@ -91,6 +96,15 @@ func newCloud(config io.Reader) (cloudprovider.Interface, error) {
 			return nil, fmt.Errorf("%s: Network %s not found", op, v)
 		}
 		networkID = n.ID
+		if networkDisableAttachedCheck := os.Getenv(hcloudNetworkDisableAttachedCheckENVVar); networkDisableAttachedCheck != "true" {
```

Comment:
We otherwise use `getEnvBool`, so we can give the user feedback on ambiguous configuration. I guess we should use it here as well.